Applications/Terminal/main.cpp

```diff
@@ -207,6 +207,19 @@ static RefPtr<GUI::Window> create_settings_window(TerminalWidget& terminal)
         terminal.set_scroll_length(value);
     };
 
+    auto& history_size_spinbox_container = settings.add<GUI::GroupBox>("Maximum scrollback history lines");
+    history_size_spinbox_container.set_layout<GUI::VerticalBoxLayout>();
+    history_size_spinbox_container.layout()->set_margins({ 6, 16, 6, 6 });
+    history_size_spinbox_container.set_size_policy(GUI::SizePolicy::Fill, GUI::SizePolicy::Fixed);
+    history_size_spinbox_container.set_preferred_size(100, 46);
+
+    auto& history_size_spinbox = history_size_spinbox_container.add<GUI::SpinBox>();
+    history_size_spinbox.set_range(512, 40960);
+    history_size_spinbox.set_value(terminal.max_history_size());
+    history_size_spinbox.on_change = [&terminal](int value) {
+        terminal.set_max_history_size(value);
+    };
```

Comment:
This is a little awkward, trying to type in the spinbox (because no one will click the buttons 9000 times) will:
- reset to 512 if you try to type a digit
- reset to 40960 if you try to clear
- *not* care if you go above 40960 (!?)

I'd rather these corrections happened on focus out instead...